server/src/test/java/io/crate/interval/IntervalParserTest.java

```diff
@@ -246,11 +260,17 @@ public void test_invalid_duplicate_units() {
         assertThatThrownBy(() -> IntervalParser.apply("1 years 2 mons 3 days 2 years"))
             .isExactlyInstanceOf(IllegalArgumentException.class)
             .hasMessage("Invalid interval format: 1 years 2 mons 3 days 2 years");
+        assertThatThrownBy(() -> IntervalParser.apply("5 millis 1 years 2 mons 3 days 2 ms"))
+            .isExactlyInstanceOf(IllegalArgumentException.class)
+            .hasMessage("Invalid interval format: 5 millis 1 years 2 mons 3 days 2 ms");
+        assertThatThrownBy(() -> IntervalParser.apply("1 years 2 mons 3 days 2.2 secs 2 ms"))
+            .isExactlyInstanceOf(IllegalArgumentException.class)
+            .hasMessage("Invalid interval format: 1 years 2 mons 3 days 2.2 secs 2 ms");
```

Comment:
Second assertion doesn't correlate with the test name `test_invalid_duplicate_units`. 
Let's remove it.  

1 years 2 mons 3 days 2.2 secs 2 ms" actually should be fine. 
I haven't checked locally, but unexpected `IllegalArgumentException` here seems to be related to `parseInteger` usage instead of `parseMilliSeconds(valueToken)` in `PGIntervalParser` (see also comment above)

Reply:
Once 2.2 seconds is specified, 200 ms are implied in the underlying Period instance, because the ms are treated in Postgres like part of the seconds. ie 2.2s evaluates to a Period with 2s and 200ms, and then ms are set and another mention of ms would trigger the invalid argument exception.

Postgres handles it the same:
```
teslamate=# select '1 year 2 months 3 weeks 5 hours 6 minutes 7.7 seconds 1000 milliseconds'::interval;
ERROR:  invalid input syntax for type interval: "1 year 2 months 3 weeks 5 hours 6 minutes 7.7 seconds 1000 milliseconds"
LINE 1: select '1 year 2 months 3 weeks 5 hours 6 minutes 7.7 second...
               ^
teslamate=# select '7 seconds 100 milliseconds'::interval;
  interval
------------
 00:00:07.1
(1 row)

teslamate=# select '7.1 seconds 100 milliseconds'::interval;
ERROR:  invalid input syntax for type interval: "7.1 seconds 100 milliseconds"
LINE 1: select '7.1 seconds 100 milliseconds'::interval;
               ^
```

Reply:
Yes, fractional sec + ms == twice ms, thanks for checking.

Maybe worth adding a comment for the second assertion as it looks confusing/not obvious.

Reply:
✅ Added comment on the assertion.